File: javascript/fizzbuzz2.js
function fizzbuzz(n, index=1, answer = []){
  // recursion
  
  if (index>n) return answer
  // console.log(index) 

  // push "Fizz"/"Buzz"/"FizzBuzz" if divisible or number.toString() to the answer array
  answer.push(
    (
      // if 3 then "Fizz" string + none or "Buzz" string if 5
      (index % 3 === 0 ? "Fizz" : "") + (index % 5 === 0 ? "Buzz" : "")
    )  
    || 
    index.toString()
  )

  return fizzbuzz(n, index+1, answer)
}

console.log(fizzbuzz(15))
// console.log(fizzbuzz(365))

File: python/fizzbuzz2.py
# def fizzbuzz(n: int):
#   answer=[]
#   for i in range (1, n+1):
#     fizz = "Fizz" * (i%3 == 0)
#     buzz = "Buzz" * (i%5 == 0)
#     answer.append( (fizz + buzz) or str(i) )
#   return answer
# # print(fizzbuzz("hello")) #errors at for loop, so type hints (m:int) is just a suggestion, good to know

def fizzbuzz(n:int):
  # list comprehension
  # ditch loops and create list with a shorter syntax
  return [
    ("Fizz" * (i%3==0) + "Buzz" * (i%5==0) or str(i))
    for i in range (1, n+1)
  ]


print(fizzbuzz(15))
